runtime/Dart/lib/src/atn/src/atn_deserializer.dart

```diff
@@ -266,39 +267,39 @@ class ATNDeserializer {
   void readRules(ATN atn) {
     final nrules = readInt();
     if (atn.grammarType == ATNType.LEXER) {
-      atn.ruleToTokenType = List<int>(nrules);
+      atn.ruleToTokenType =
+          List<int>.filled(nrules, 0); //Todo: keep an eye on this zero
```

Comment:
is it ok to do this?

Reply:
It matches Java behaviour:
https://github.com/antlr/antlr4/blob/b5648f405b746d87ec559132ad1939fb6d402053/runtime/Java/src/org/antlr/v4/runtime/atn/ATNDeserializer.java#L263

Reply:
Changed this to grow the list as iterating until nrules. Essentially the same thing without adding the temporary zeros.